### DISCUSSION TITLE
Making a child window behave properly.

### DISCUSSION THREAD

**Discussion by jtreagan** [fltk-rs/fltk-rs#1557] [Q&A] [answered]
This morning I've been playing around with adding child windows to a main window.  The app doesn't do anything.  I'm just using it to experiment with how child and parent windows relate.  I need to get comfortable with this so that I can make FLTK work right for me.
```
use fltk::{app, prelude::*, window::Window};
use fltk::enums::Color;
use fltk::frame::Frame;

fn main() {
    let app = app::App::default();
    let mut mainwin = Window::default()
        .with_size(500, 500)
        .with_label("Main Window")
        .center_screen();

    let mut win2 = Window::new(10, 10, 100, 100, "Window #2");
    win2.set_color(Color::Red);
    win2.make_resizable(true);
    let _frame = Frame::new(10, 10, 100, 100, "Window #2");
    win2.show();
    win2.end();

    mainwin.show();
    mainwin.end();

    app.run().unwrap();
}
```
When I run this in the terminal three problems immediately present themselves:
1) The child window (win2) doesn't display the indicated label or any window controls.  (I added a frame so that I could have some kind of label on the window.)
2) The child window isn't resizable.
3) The child window isn't movable.

So far I haven't found a solution, so maybe someone can point out to me what I'm doing wrong?  Thanks.

BTW: Just in case it matters, I'm using PopOS linux 22.04.

**Comment by MoAlyousef** [comment, marked as answer]
A child window is embedded in its parent. So it won't have a label since a window's label is displayed on its decoration (border), and an embedded window is borderless.
Also it's embedded in place, so it can't be moved unless you specify how it should move in the handle method.

Basically an embedded window is more similar to a Group widget than a Window. It has the advantage though of having a native handle which you can pass to other libraries for them to draw on it.
